Timeout transition action

JoshuaCarter opened the pull request:
@nyeholt For your review. Biggest question is whether default-jobs config should be just in readme or also in config.yml

Commit:
add(TimeoutTransition): Action to auto transition on timeout

Commit:
add(TimeoutJob): Job to process timeout actions

Commit:
add(config): Config for default WorkflowTimeoutJob

nyeholt commented on src/Actions/TimeoutTransitionAction.php:
```diff
@@ -0,0 +1,60 @@
+
+        // increments
+        $inc = [];
+        preg_match('/Enum\(\"(.*)\"\)/', static::$db['TimeoutIncrement'], $inc);
```
For future ref, you can access these values as

$increments = $this->dbObject('TimeoutIncrement')->enumValues()

JoshuaCarter replied:
Fixed.

nyeholt commented on src/Jobs/WorkflowTimeoutJob.php:
```diff
@@ -0,0 +1,55 @@
+
+    public function setup()
+    {
+        $ttis = TimeoutTransitionInstance::get();
```
Will this also include the transition  instances for completed workflows?

JoshuaCarter replied:
True, added obvious filter.

Commit:
add(WorkflowTimeoutJob): Don't process finished transitions

Commit:
chore(TimeoutTransitionAction): Use enumValues() over regex

nyeholt commented on src/Actions/TimeoutTransitionInstance.php:
```diff
@@ -0,0 +1,83 @@
+        if ($time = $this->getTimeoutTime()) {
+            return date($format, $time);
+        }
+        return 'never';
```
Does "never" actually mean anything in this context? Or is this what gets shown to the user?

nyeholt commented on src/Actions/TimeoutTransitionInstance.php:
```diff
@@ -0,0 +1,83 @@
+            'Comment' => 'Automatically transitioned due to timeout'
+        ]);
+
+        return false;
```
Just from a readability point of view, having this return false is a little confusing when the other return values are an error message. I'd just let it exit (or explicitly return null) because a 'false' makes it sound like the transition attempt failed.

JoshuaCarter replied:
Refactored this to return true/false as success/failure, and a ref param for err string.

nyeholt commented:
Let's remove the config and leave it as documentation only; it feels like something that should be up to the project developer to add in, rather than forcing it in.

JoshuaCarter commented:
Config removed.

Commit:
fix(TimeoutTransitionAction): Execute should return false (and misc)

- Execute returns false so that Finished is false (allows filtering).
- Minor non-functional refactor of attemptTimeoutTransition returns.
- Remove config for WorkflowTimeoutJob (still in readme).
